packages/squiggle-lang/src/public/SqProject/SqModuleOutput.ts

```diff
@@ -0,0 +1,270 @@
+import { isBindingStatement } from "../../ast/utils.js";
+import { Env } from "../../dists/env.js";
+import { RunProfile } from "../../reducer/RunProfile.js";
+import { BaseRunner, RunParams } from "../../runners/BaseRunner.js";
+import { ImmutableMap } from "../../utility/immutable.js";
+import { Err, fmap, fmap2, Ok, result } from "../../utility/result.js";
+import { vDict, VDict } from "../../value/VDict.js";
+import { vString } from "../../value/VString.js";
+import {
+  SqError,
+  SqImportError,
+  SqOtherError,
+  wrapIError,
+} from "../SqError.js";
+import { SqValue, wrapValue } from "../SqValue/index.js";
+import { SqDict } from "../SqValue/SqDict.js";
+import { SqValueContext } from "../SqValueContext.js";
+import { SqValuePath, ValuePathRoot } from "../SqValuePath.js";
+import { Externals, RunContext } from "./ProjectItem.js";
+import { ProjectState } from "./ProjectState.js";
+import { SqModule } from "./SqModule.js";
+import { getHash } from "./utils.js";
+
+export type OutputResult = result<
+  {
+    result: SqValue;
+    bindings: SqDict;
+    imports: SqDict;
+    exports: SqDict;
+    profile: RunProfile | undefined;
+  },
+  SqError
+>;
+
+export class SqModuleOutput {
+  module: SqModule;
+  environment: Env;
+  result: OutputResult;
+  executionTime: number;
```

Comment:
This field is now here and not in components `Simulation` type.

In general, `SqModuleOutput` replaces several similar but not identical types we had before.